src/Umbraco.Web/Mvc/RenderRouteHandler.cs

```diff
@@ -279,10 +280,40 @@ internal virtual RouteDefinition GetUmbracoRouteDefinition(RequestContext reques
                     //set the controller and name to the custom one
                     def.ControllerType = controllerType;
                     def.ControllerName = ControllerExtensions.GetControllerName(controllerType);
+
                     if (def.ControllerName != defaultControllerName)
                     {
                         def.HasHijackedRoute = true;
                     }
+
+                    // Look for hijacked action
+                    var hijackActions = controllerType
+                        .GetMethods(BindingFlags.Public | BindingFlags.Instance)
+                        .Select(x => Tuple.Create(x, x.GetCustomAttribute<HijackRouteAttribute>(true)))
+                        .Where(x => x.Item2 != null && x.Item2.ContentTypeAlias.InvariantEquals(request.PublishedContent.ContentType.Alias))
+                        .ToList();
+
+                    if (hijackActions.Count > 0)
+                    {
+                        MethodInfo hijackAction = null;
+
+                        if (request.HasTemplate)
+                        {
+                            hijackAction = hijackActions
+                                .FirstOrDefault(x => x.Item2
```

Comment:
Since it will be possible to add the same attributes with the same template alias/action name to multiple actions, using FirstOrDefault here will mean developers don't really know which one would execute. 
Do we need to consider what should happen in this scenario? Is there a case where a developer would want/need multiple attributes with the same values? Else should an exception throw to inform a developer that they have duplicates?